### PR TITLE
Add more modifier checks to MethodDeclaration node

### PR DESCRIPTION
**Pull request by inxilpro**

### CODE REVIEW

**Comment by roblourens** [comment]
Thanks for the PR! Can you tell me about why you want this change or what you are using this parser for?

**Comment by inxilpro** [comment]
I'm working on a [PHP linter](https://github.com/glhd/laralint) that uses this parser, and I need to be able to query the nodes more easily. I originally was decorating the nodes, but decided to try a PR to see how receptive you'd be to this kind of improvement. Sorry about the really sparse PR!

File: src/Node/MethodDeclaration.php [hunk 1]
```diff
@@ -40,18 +40,38 @@ class MethodDeclaration extends Node implements FunctionLike {
...
-            if ($modifier->kind === TokenKind::StaticKeyword) {
-                return true;
-            }
-        }
-        return false;
+        return $this->hasModifier(TokenKind::StaticKeyword);
     }
+	
+	public function isPublic() : bool {
+		return $this->hasModifier(TokenKind::PublicKeyword);
```
Comment (TysonAndre) [inline src/Node/MethodDeclaration.php]: `class X { function test() {}}` still has an implicitly public method `X::test()` - I can see this being misunderstood in applications using this helper, for that edge case.

This is low priority for me, though

Comment (roblourens) [reply src/Node/MethodDeclaration.php]: Yeah that's a good point. Could change `isPublic` to just check `!isPrivate` but it might be better to just expose `hasModifier`, I think the goal is to expose the AST but not necessarily "interpret" it...

Comment (TysonAndre) [reply src/Node/MethodDeclaration.php]: It might be useful to have a link to projects using tolerant-php-parser in the documentation, as examples for people writing their own applications/libraries.

E.g. there's https://github.com/tysonandre/tolerant-php-parser-to-php-ast (part of Phan), https://github.com/felixfbecker/php-language-server , etc.

The first one is notable in trying to convert all of the information provided by various node types.

Comment (roblourens) [reply src/Node/MethodDeclaration.php]: I made that change and just left `hasModifier`, FYI @inxilpro

Comment (inxilpro) [reply src/Node/MethodDeclaration.php]: I could also see changing these to `hasPublicModifier()`/etc to be more clear about what question it's answering.
